service/src/test/java/com/epam/rft/atsy/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,188 @@
+import org.springframework.dao.EmptyResultDataAccessException;
+
+import javax.persistence.NoResultException;
+import javax.persistence.NonUniqueResultException;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.mockito.BDDMockito.given;
+
+@RunWith(MockitoJUnitRunner.class)
+public class UserServiceImplTest {
+
+    private static final long USER_ID = 1L;
+    private static final String USER_NAME = "Test";
+    private static final String USER_PASSWORD = "password";
+
+    private static final String NOT_REAL_USER_NAME = "notreal";
+    private static final String NULL_USER_NAME = null;
+
+    private static final UserDTO NULL_USER_DTO = null;
+    private static final UserEntity NULL_USER_ENTITY = null;
+
+    private UserDTO userDTO;
+    private UserDTO detailedUserDTO;
+    private UserEntity userEntity;
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private ModelMapper modelMapper;
+
+    @InjectMocks
+    private UserServiceImpl userService;
+
+    @Before
+    public void setUp() {
+        userDTO = UserDTO.builder().name(USER_NAME).password(USER_PASSWORD).build();
+        detailedUserDTO = UserDTO.builder().id(USER_ID).name(USER_NAME).password(USER_PASSWORD).build();
+        userEntity = UserEntity.builder().id(USER_ID).userName(USER_NAME).userPassword(USER_PASSWORD).build();
+
+        given(modelMapper.map(userEntity, UserDTO.class)).willReturn(detailedUserDTO);
+        given(modelMapper.map(detailedUserDTO, UserEntity.class)).willReturn(userEntity);
+    }
+
+    @Test(expected = UserNotFoundException.class)
+    public void loginShouldThrowUserNotFoundExceptionBecauseOfNoResultException() throws UserNotFoundException {
+        //Given
+        given(userRepository.findByUserNameAndUserPassword(USER_NAME, USER_PASSWORD)).willThrow(NoResultException.class);
+
+        //When
+        userService.login(userDTO);
+    }
+
+    @Test(expected = UserNotFoundException.class)
+    public void loginShouldThrowUserNotFoundExceptionBecauseOfEmptyResultDataAccessException() throws UserNotFoundException {
+        //Given
+        given(userRepository.findByUserNameAndUserPassword(USER_NAME, USER_PASSWORD)).willThrow(EmptyResultDataAccessException.class);
+
+        //When
+        userService.login(userDTO);
+    }
+
+    @Test(expected = UserNotFoundException.class)
+    public void loginShouldThrowUserNotFoundExceptionBecauseOfNonUniqueResultException() throws UserNotFoundException {
+        //Given
+        given(userRepository.findByUserNameAndUserPassword(USER_NAME, USER_PASSWORD)).willThrow(NonUniqueResultException.class);
+
+        //When
+        userService.login(userDTO);
+    }
+
+    @Test(expected = BackendException.class)
+    public void loginShouldThrowBackendExceptionBecauseOfNullPointerException() throws UserNotFoundException {
```

Comment:
Please use Assert.notNull() in login method to be consistent with saveOrUpdate method (and other service implementations which check their parameters).